src/femtomes.cpp

```diff
@@ -0,0 +1,527 @@
+			crc = calculateBlockCRC32((uint32_t)_femto_msg.header.femto_header.messagelength, (uint8_t *)&_femto_msg.data[0], crc);
+
+			if (_femto_msg.crc == crc)
+			{
+				iRet = _femto_msg.read;
+			}
+			else
+			{
+				FEMTO_DEBUG("data packet is bad");
+			}
+			break;
+	}
+	return iRet;
+}
+
+void GPSDriverFemto::decodeInit()
+{
+	_decode_state = FemtoDecodeState::pream_ble1;
+
+	if (_output_mode == OutputMode::RTCM) {
+		if (!_rtcm_parsing) {
+			_rtcm_parsing = new RTCMParsing();
+		}
+		_rtcm_parsing->reset();
+	}
+
+}
+
+int GPSDriverFemto::writeAckedCommandFemto(const char* command, const char* reply, const unsigned int timeout)
+{
+	/**< write command*/
+	write(command, strlen(command));
+	/**< wait for reply*/
+	uint8_t buf[GPS_READ_BUFFER_SIZE];
+	gps_abstime time_started = gps_absolute_time();
+	while (time_started + timeout * 1000 * 2 > gps_absolute_time()) {
+		int ret = read(buf, sizeof(buf), 1000); /**< wait 1000us */
+		buf[sizeof(buf)-1] = 0;
+		if (ret > 0 && strstr((char *)buf, reply) != NULL) {
+			FEMTO_DEBUG("command reply success: %s", command);
+			return 0;
+		}
+	}
+	return -1;
+}
+
+int GPSDriverFemto::configure(unsigned &baudrate,OutputMode output_mode)
+{
+	_output_mode = output_mode;
+	_correction_output_activated = false;
+	_configure_done = false;
+
+	/** Try different baudrates (115200 is the default for Femtomes) and request the baudrate that we want.	 */
+	const unsigned baudrates_to_try[] = {115200};
+	bool success = false;
+
+	unsigned test_baudrate;
+
+	for (unsigned int baud_i = 0; !success && baud_i < sizeof(baudrates_to_try) / sizeof(baudrates_to_try[0]); baud_i++) {
+		test_baudrate = baudrates_to_try[baud_i];
+
+		if (baudrate > 0 && baudrate != test_baudrate) {
+			continue; /**< skip to next baudrate*/
+		}
+
+		setBaudrate(test_baudrate);
+
+		FEMTO_DEBUG("baudrate set to %i", test_baudrate);
+
+		for (int run = 0; run < 2; ++run) { /** try several times*/
+			if (writeAckedCommandFemto("UNLOGALL\r\n", "<UNLOGALL OK", FEMO_RESPONSE_TIMEOUT) == 0 &&
+				writeAckedCommandFemto("VERSION\r\n", "<VERSION OK", FEMO_RESPONSE_TIMEOUT) == 0) {
+				FEMTO_DEBUG("got port for baudrate %i", test_baudrate);
+				success = true;
+				break;
+			}
+		}
+	}
+
+	if (!success) {
+		FEMTO_DEBUG("femtomes gps start failed %i", test_baudrate);
+		return -1;
+	}
+	/**
+	* We successfully got a response and know to which port we are connected. Now set the desired baudrate
+	* if it's different from the current one.
+	*/
+	const unsigned desired_baudrate = 115200; /**< changing this requires also changing the SPD command*/
+
+	baudrate = test_baudrate;
+
+	if (baudrate != desired_baudrate) {
+		baudrate = desired_baudrate;
+		const char baud_config[] = "com%c 115200\r\n"; // configure baudrate to 115200
+		char baud_config_str[sizeof(baud_config)];
+		int len = snprintf(baud_config_str, sizeof(baud_config_str), baud_config, _port);

```

Comment:
`_port` is just a space (' '), so this can be simplified. Did you intend to make this configurable?

Reply:
thanks, delete this already.